tls/s2n_quic_support.c

```diff
@@ -55,3 +58,39 @@ int s2n_connection_get_quic_transport_parameters(struct s2n_connection *conn,
 
     return S2N_SUCCESS;
 }
+
+/* When using QUIC, S2N reads unencrypted handshake messages instead of encrypted records.
+ * This method sets up the S2N input buffers to match the results of using s2n_read_full_record.
+ */
+int s2n_quic_read_handshake_message(struct s2n_connection *conn, uint8_t *message_type)
+{
+    notnull_check(conn);
+
+    /* Allocate the maximum stuffer size now so that we don't have to realloc later in the handshake. */
+    GUARD(s2n_stuffer_resize_if_empty(&conn->in, S2N_MAXIMUM_HANDSHAKE_MESSAGE_LENGTH));
```

Comment:
Does this mean each connection will require at least 128k of memory (one for in, one for out)? This would be a good argument for reusing connection structs as much as possible (assuming this buffer is wiped and not freed).